mautic/mautic
# Form kiosk mode

Many vendors want to be able to use a Mautic form to have leads submit information at events, office, etc. The problem is that Mautic would assume that the form is being submitted from the lead's device so the IP address would be assigned to the lead, tracking cookies generated, sometimes leads would merge, etc.  

So in comes kiosk mode.  When a form is set to be in kiosk mode, it will not assign the IP address to the lead (as it's assumed it's the vendors IP and not the lead's), it will not generate new tracking cookies, it will not merge the current lead with the newly created or updated lead from the form (although it will update a lead if one is found with an existing email).

To test the patch:

1)  run `php app/console doctrine:schema:update --force` to add the new in_kiosk_mode column to the form table.
2) Create a new form with at least an email field, a create/update lead submit action, and switch the toggle "Kiosk Mode" to yes.
3) Now open a separate browser and go to the public URL for the form (http://your.mauticurl.com/form/1 where 1 is the id of your form). 
4) Look at your cookies and you will see mautic_session_id with a random string value.  Find the cookie that has a name that is the same as that random string.  It's value will be the lead ID. This is the vendor and likely an anonymous lead.  Mautic creates these leads way early in the process before the form stuff is reached so these cookie are expected.
5) Submit the form with an email that does not already exist.

After submitting, check the cookies.  They should be the same.  Then check your leads in Mautic to ensure your new lead was created and note it's ID.  Finally look at the following db tables:

form_submissions should have the lead_id assigned but not tracking_id
lead_lead_ips_xref should not have the lead_id assigned to the local IP

Try submitting a few more submissions.  They should all have the same results as above.  

Also try submitting a few with emails of existing leads. Those leads should be updated per the create/update lead submit action settings but you should not have leads merging if emails do not match.

Now disable kiosk mode and repeat the same.  This time, if your current lead cookie is associated with an anonymous lead, you'll find that your first form submit will result in that lead being updated with the submitted details. The local IP will be assigned to the lead and a tracking id in the submission table.  If submitting another form with an email that doesn't already exist, you should see a new lead (NOT merged into the previous), the ip address assigned to that lead as well, and new cookies generated.
